connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -0,0 +1,277 @@
+                inner = new TreeMap<>();
+                pluginLoaders.put(pluginClassName, inner);
+            }
+            inner.put(plugin, loader);
+        }
+    }
+
+    public void initLoaders() {
+        for (String path : pluginPaths) {
+            try {
+                Path pluginPath = Paths.get(path).toAbsolutePath();
+                if (Files.isDirectory(pluginPath)) {
+                    for (Path dir : pluginDirs(pluginPath)) {
+                        log.info("Loading dir: {}", dir);
+                        URL[] jars = jarPaths(dir).toArray(new URL[0]);
+                        // log.info("Loading jars: " + Arrays.toString(jars));
+                        PluginClassLoader loader = newPluginClassLoader(jars);
+                        log.info("Using loader: {}", loader);
+                        PluginScanResult plugins = scanPluginPath(loader, jars);
+
+                        if (plugins.isEmpty()) {
+                            inactivePaths.put(dir, null);
+                        } else {
+                            activePaths.put(dir, loader);
+                        }
+
+                        addPlugins(plugins.connectors(), loader);
+                        connectors.addAll(plugins.connectors());
+                        addPlugins(plugins.converters(), loader);
+                        converters.addAll(plugins.converters());
+                        addPlugins(plugins.transformations(), loader);
+                        transformations.addAll(plugins.transformations());
+                    }
+                }
+            } catch (InvalidPathException | MalformedURLException e) {
+                log.warn("Invalid path in plugin path: {}. Ignoring.", path);
+            } catch (IOException e) {
+                log.warn("Could not get listing for plugin path: {}. Ignoring.", path);
+            } catch (InstantiationException | IllegalAccessException e) {
+                log.warn("Could not instantiate plugins in: {}. Ignoring: {}", path, e);
+            }
+        }
+    }
+
+    private static PluginClassLoader newPluginClassLoader(final URL[] jars) {
+        return (PluginClassLoader) AccessController.doPrivileged(
+                new PrivilegedAction() {
+                    @Override
+                    public Object run() {
+                        return new PluginClassLoader(jars);
+                    }
+                }
+        );
+    }
+
+    public Set<PluginDesc<Connector>> connectors() {
+        return connectors;
+    }
+
+    public Set<PluginDesc<Converter>> converters() {
+        return converters;
+    }
+
+    public Set<PluginDesc<Transformation>> transformations() {
+        return transformations;
+    }
+
+    private PluginScanResult scanPluginPath(
+            PluginClassLoader loader,
+            URL[] urls
+    ) throws InstantiationException, IllegalAccessException {
+        //log.info("Scanning plugin path urls: " + Arrays.toString(urls));
```

Comment:
For sure. It was obstructing me during debugging. Also more `debug` and `trace` logs are needed.